Frameworks/Foundation/_NSUndoGroup.mm

```diff
@@ -0,0 +1,91 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import <Foundation/NSUndoManager.h>
+#import "_NSUndoManagerInternal.h"
+
+@implementation _NSUndoGroup {
+    StrongId<_NSUndoManagerStack> _undoGrouping;
```

Comment:
>_undoGrouping; [](start = 34, length = 14)

not sure how I feel about this variable name basically being a repeat of the class name - got me confused several times